multiversx_sdk_cli/governance.py

```diff
@@ -0,0 +1,256 @@
+from multiversx_sdk import (
+    Address,
+    GovernanceTransactionsFactory,
+    Transaction,
+    TransactionsFactoryConfig,
+    VoteType,
+)
+
+from multiversx_sdk_cli.base_transactions_controller import BaseTransactionsController
+from multiversx_sdk_cli.guardian_relayer_data import GuardianRelayerData
+from multiversx_sdk_cli.interfaces import IAccount
+
+
+class GovernanceWrapper(BaseTransactionsController):
+    def __init__(self, config: TransactionsFactoryConfig) -> None:
+        self.factory = GovernanceTransactionsFactory(config)
+
+    def create_transaction_for_new_proposal(
+        self,
+        sender: IAccount,
+        nonce: int,
+        commit_hash: str,
+        start_vote_epoch: int,
+        end_vote_epoch: int,
+        native_token_amount: int,
+        gas_limit: int,
+        gas_price: int,
+        version: int,
+        options: int,
+        guardian_and_relayer_data: GuardianRelayerData,
+    ) -> Transaction:
+        tx = self.factory.create_transaction_for_new_proposal(
+            sender=sender.address,
+            commit_hash=commit_hash,
+            start_vote_epoch=start_vote_epoch,
+            end_vote_epoch=end_vote_epoch,
+            native_token_amount=native_token_amount,
+        )
+        tx.nonce = nonce
+        tx.version = version
+        tx.options = options
+        tx.gas_price = gas_price
+        tx.guardian = guardian_and_relayer_data.guardian_address
+        tx.relayer = guardian_and_relayer_data.relayer_address
+
+        self.add_extra_gas_limit_if_required(tx)
+
+        if gas_limit:
+            tx.gas_limit = gas_limit
+
+        self.sign_transaction(
+            transaction=tx,
+            sender=sender,
+            guardian=guardian_and_relayer_data.guardian,
+            relayer=guardian_and_relayer_data.relayer,
+            guardian_service_url=guardian_and_relayer_data.guardian_service_url,
+            guardian_2fa_code=guardian_and_relayer_data.guardian_2fa_code,
+        )
+
+        return tx
+
+    def create_transaction_for_voting(
+        self,
+        sender: IAccount,
+        nonce: int,
+        proposal_nonce: int,
+        vote: str,
+        gas_limit: int,
+        gas_price: int,
+        version: int,
+        options: int,
+        guardian_and_relayer_data: GuardianRelayerData,
+    ) -> Transaction:
+        vote_value = self._vote_type_from_string(vote)
+
+        tx = self.factory.create_transaction_for_voting(
+            sender=sender.address, proposal_nonce=proposal_nonce, vote=vote_value
+        )
+        tx.nonce = nonce
+        tx.version = version
+        tx.options = options
+        tx.gas_price = gas_price
+        tx.guardian = guardian_and_relayer_data.guardian_address
+        tx.relayer = guardian_and_relayer_data.relayer_address
```

Comment:
I would extract this into a common set 
  tx.nonce = nonce
        tx.version = version
        tx.options = options
        tx.gas_price = gas_price
        tx.guardian = guardian_and_relayer_data.guardian_address
        tx.relayer = guardian_and_relayer_data.relayer_address
        
as it is used in each method